File: ruby/week_1/10_mongo_exception.rb
require 'mongo'

mongo_connection = Mongo::Connection.new('localhost', 27017, :safe => true)

db = mongo_connection["test"]
users = db["users"]

j = {first_name: "Andrew", last_name: "Erlichson"}

begin
  users.insert(j)
rescue
  puts "insert failed: %s" % $!
end

puts j

begin
  users.insert(j)
rescue
  puts "insert failed: %s" % $!
end

File: python/week_1/10_mongo_exception.py
import pymongo
import sys

connection = pymongo.Connection("mongodb://localhost", safe=True)

db = connection.test
users = db.users

j = {"first_name": "Andrew", "last_name": "Erlichson"}

try:
    users.insert(j)
except:
    print "insert failed:", sys.exc_info()[0]

print j

try:
    users.insert(j)
except:
    print "insert failed:", sys.exc_info()[0]
